src/embedding/ad.ts

```diff
@@ -0,0 +1,738 @@
+    }
+
+    //
+    // Drillable Items command
+    //
+
+    /**
+     * Base type of drillable items command body
+     */
+    export interface ISimpleDrillableItemsCommandBody {
+        /**
+         * The array of uris of attributes or measures
+         */
+        uris?: string[];
+        /**
+         * The array of identifiers of attributes or measures
+         */
+        identifiers?: string[];
+    }
+
+    /**
+     * The main data type for furture processing of drillable items command
+     */
+    export interface IDrillableItemsCommandBody extends ISimpleDrillableItemsCommandBody {
+        /**
+         * Master measures items - In-case, a derived measure is composed from a master measure.
+         */
+        composedFrom?: ISimpleDrillableItemsCommandBody;
+    }
+
+    /**
+     * Set drillable items.
+     *
+     * Contract:
+     *
+     * - Drillable items can be set by uris or identifiers of insight's measures/attributes
+     */
+    export type DrillableItemsCommand = IADMessageEvent<GdcAdCommandType.DrillableItems, IDrillableItemsCommandBody>;
+
+    /**
+     * Data type of drillable items command
+     *
+     * Note: The main event data was wrapped to application and product data structure
+     * @see IDrillableItemsCommandBody
+     */
+    export type DrillableItemsCommandData = IAdGdcMessageEnvelope<
+        GdcAdCommandType.DrillableItems,
+        IDrillableItemsCommandBody
+    >;
+
+    /**
+     * Type-guard checking whether an object is an instance of {@link DrillableItemsCommandData}
+     *
+     * @param obj - object to test
+     */
+    export function isDrillableItemsCommandData(obj: any): obj is DrillableItemsCommandData {
+        return getEventType(obj) === GdcAdCommandType.DrillableItems;
+    }
+
+    //
+    // Open insight command
+    //
+
+    /**
+     * Contain the information to contruct the AD url to open an insight editor
+     */
+    export interface IOpenInsightCommandBody {
+        /**
+         * Dataset identifier - A dataset consists of attributes and facts,
+         * which correspond to data you want to measure and the data
+         * that you want to use to segment or filter those measurements.
+         */
+        dataset?: string;
+
+        /**
+         * Project id
+         */
+        projectId?: string;
+
+        /**
+         * Client id - Each client has an identifier unique within the domain
+         *
+         * Note: use the combination of the data product ID and client ID instead of the project ID
+         */
+        clientId?: string;
+
+        /**
+         * Product id - A data product contains multiple segments. And a segment has clients assigned to it
+         *
+         * Note: use the combination of the data product ID and client ID instead of the project ID
+         */
+        productId?: string;
+
+        /**
+         * Insight id - leave it empty to reset the insight editor to empty state
```

Comment:
does it mean that empty insightId == AD's 'Clear' function

Reply:
yes. It same as the first time we access the AD and the url have format like
`/analyze/#/<projectId>/reportId/edit`